core/string/ustring.cpp

```diff
@@ -1644,6 +1644,33 @@ String String::hex_encode_buffer(const uint8_t *p_buffer, int p_len) {
 	return ret;
 }
 
+Vector<uint8_t> String::hex_decode() const {
+	ERR_FAIL_COND_V_MSG(length() % 2 != 0, Vector<uint8_t>(), "Hexadecimal string of uneven length.");
+
+#define HEX_TO_BYTE(c, o)                                                                                                                     \
+	if (is_digit(c)) {                                                                                                                        \
+		*o = c - '0';                                                                                                                         \
+	} else if (c >= 'a' && c <= 'f') {                                                                                                        \
+		*o = c - 'a' + 10;                                                                                                                    \
+	} else if (c >= 'A' && c <= 'F') {                                                                                                        \
+		*o = c - 'A' + 10;                                                                                                                    \
+	} else {                                                                                                                                  \
+		ERR_FAIL_V_MSG(Vector<uint8_t>(), "Invalid hexadecimal notation character \"" + chr(c) + "\" in part \"" + substr(i * 2, 2) + "\"."); \
```

Comment:
I don't think it makes sense to print part of a string (because it is homogeneous, it has no delimiters). I think it's enough just to specify the first invalid character (and probably its position, since the character may be non-printable).